### PR TITLE
feat: endpoint for checking if document exist

### LINKED ISSUE #529: feat: add endpoint to check that an address exist or not
- [ ] return 200 and 404 or just use 200 and reply response

### PR DESCRIPTION
**Pull request by collinlokken**
## What does this pull request change?
Introduces an endpoint for checking if a document exists. The endpoint returns a boolean based on a document address, and a 200 OK for both cases. 

## Why is this pull request needed?
Gives the ability to check if a document exists without creating error logs.

## Issues related to this change:
Closes #529

### CODE REVIEW

File: src/tests/unit/test_remove.py [hunk 1]
```diff
@@ -20,7 +19,7 @@ def setUp(self):
         self.document_service = get_mock_document_service(lambda x, y: self.repository)
 
     def mock_get(self, document_id: str):
-        return deepcopy(self.storage[document_id])
```
Comment (collinlokken) [inline src/tests/unit/test_remove.py:23]: automatically removed by pre-commit. I thought it was a bit strange, but...

File: src/features/document/document_feature.py [hunk 2]
```diff
@@ -174,3 +175,18 @@ def remove(address: str, user: User = Depends(auth_w_jwt_or_pat)):
     - str: "OK" (200)
     """
     return remove_use_case(user=user, address=address)
+
+
+@router.get("-existence/{address:path}", operation_id="document_check", response_model=bool, responses=responses)
```
Comment (KristianKjerstad) [inline src/features/document/document_feature.py:180]: is it possible to define the endpoint to be /address/exists? Not sure if fast api is able to interpret the url parameter address in this case. But the url /documents/address/exists is a bit easier to read than /documents-existence/address IMO
```suggestion
@router.get("{address:path}/exists", operation_id="document_check", response_model=bool, responses=responses)
```

File: src/features/document/document_feature.py [hunk 3]
```diff
@@ -174,3 +175,18 @@ def remove(address: str, user: User = Depends(auth_w_jwt_or_pat)):
...
+@create_response(JSONResponse)
+def check_existence(address: str, user: User = Depends(auth_w_jwt_or_pat)):
+    """Checks if a document exists, given an address.
```
Comment (KristianKjerstad) [inline src/features/document/document_feature.py:183]:
```suggestion
    """Checks if an entity exists, given an address.
```

when address is datasource/path/car.engine, the address points to an entity and not a document. This is a definition thing, but all documents are actually entities (but not all entities are documents).

Documents are basically what is stored in mongo db as a separate "entry"

File: src/tests/bdd/document/check_existence.feature [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      "content": [],
+      "isRoot": true
+    }
+    """
+
+  Scenario: Check the existence of non-existing document
+    Given I access the resource url "/api/documents-existence/test-DS/$1"
+    When I make a "GET" request
+    Then the response status should be "OK"
+    And the response should contain
```
Comment (KristianKjerstad) [inline src/tests/bdd/document/check_existence.feature:24]:
```suggestion
    And the response should be
```
better to use the "... should be" function if we know the exact result the endpoint should return

File: src/tests/bdd/document/check_existence.feature [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    """
+    true
+    """
+    Given I access the resource url "/api/documents-existence/test-DS/$2"
+    When I make a "GET" request
+    Then the response status should be "OK"
+    And the response should contain
```
Comment (KristianKjerstad) [inline src/tests/bdd/document/check_existence.feature:31]:
```suggestion
    And the response should be
```
better to use the "... should be" function if we know the exact result the endpoint should return

File: src/tests/unit/test_check_existence.py [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    check_existence_use_case,
+)
+from tests.unit.mock_utils import get_mock_document_service
+
+
+class CheckExistenceTestCase(unittest.TestCase):
+    def setUp(self) -> None:
+        self.storage = {
+            "1": {
+                "name": "Test document",
```
Comment (KristianKjerstad) [inline src/tests/unit/test_check_existence.py:16]:
```suggestion
                "name": "Test_document",
```
We have defined a rule that name should not contain spaces (see EntityName class in shared.py). Not sure if we have added validation for that...

**Review by KristianKjerstad — COMMENTED** [review]
Nice!

A few suggestions for improvement.

**Review by KristianKjerstad — APPROVED** [review]
🎉

**Pull request merged**